src/error.rs

```diff
@@ -0,0 +1,44 @@
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub struct TokenizersError(pub tokenizers::Error);
+
+impl PartialEq for TokenizersError {
+    fn eq(&self, other: &Self) -> bool {
+        self.0.to_string() == other.0.to_string()
+    }
+}
+
+impl std::fmt::Display for TokenizersError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+#[derive(Error, Debug, PartialEq)]
+pub enum Error {
+    #[error("The vocabulary does not allow us to build a sequence that matches the input")]
+    IndexError,
```

Comment:
We might have other `Index` errors so it might be better to use `VocabularyIsInsufficientToBuildIndex` would be a better name for this variant.

Reply:
This error doesn't belong to vocabulary. It's from `Index` interface, which I'm not working on right now. When we'll be adjusting `Index` it will be easier to see what other errors varieties we need.

Reply:
As you wish, I just wanted to mention that the name of the error variant doesn't match the error message. But if it's planned to be addressed in a later PR, that's totally fine!

Reply:
Yeah, I appreciate you mentioned that too, since it will be in following PRs for sure 👍